compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1761,14 +1752,6 @@ fn receiver_is_valid<'tcx>(
                 // We cannot proceed.
                 break;
             }
```

Comment:
```suggestion
            }

            // Register the bound, in case it has any region side-effects.
            wfcx.register_bound(
                cause.clone(),
                wfcx.param_env,
                potential_self_ty,
                receiver_trait_def_id,
            );
```